.github/workflows/release-drafter.yml

```diff
@@ -34,10 +37,11 @@ jobs:
         run: |
           python -m build
           tar -zvcf artifacts.tar.gz dist
+      - name: Publish package distributions to PyPI
```

Comment:
Aren't we doing the same thing [here](https://github.com/opensearch-project/opensearch-py-ml/blob/c9779818ce3d1e16916891d182dcadcdd22f0f94/jenkins/release.JenkinsFile#L12)?

Reply:
Yes. The reason we did in Jenkins was that we signed our GPG signatures in Jenkins workflow. Since Pypi now support 2FA/trusted publisher and doesn't require GPG signatures anymore. There is no need for us to run Jenkins for release. 

Therefore, we could simply use GHA for release. https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-pypi

Reply:
In that case don't we need to remove the jenkins workflow code? or it will trigger both workflow?

Reply:
Yes. We could remove the jenkins workflow or we can disable on Jenkins side as well. Maybe let's keep the jenkins code for now until we successfully release next time through GHA?

Reply:
I'm planning to make a release in next month. So I just wanted to make sure if I try to release an artifact two workflow won't trigger together, that's all. If you can confirm, that won't be the case, I'm fine to approve this PR. Thanks.

Reply:
Yes, I will disable the jenkins workflow to avoid it triggered. 
For this GHA release workflow, I will need to add trusted publisher within our Pypi account to make sure it will run. 

I will make sure complete two of above tasks so that only GHA workflow will run for release once this is merged. Thanks.